tests/js/client/aql/aql-optimizer-indexes-group-sort.js

```diff
@@ -191,24 +191,174 @@ function optimizerIndexesGroupSortTestSuite() {
+        let plan = query_plan(query, collection.name());
+        assertTrue(query_plan_uses_index_for_sorting(plan), plan.rules);
+        assertTrue(sort_node_does_a_group_sort(plan), plan.nodes);
+        assertEqual(expected_results(query, collection.name()), execute(query, collection.name()), query);
+      }
+
+      collection.ensureIndex({ type: "persistent", fields: ["c", "d"], sparse: true });
+      waitForEstimatorSync();
+
+      query = "FOR doc IN @@collection FILTER doc.c > null AND doc.d > null SORT doc.c, doc.d, doc.x RETURN doc.b";
+      plan = query_plan(query, collection.name());
       assertTrue(query_plan_uses_index_for_sorting(plan), plan.rules);
-      assertTrue(sort_node_does_a_group_sort(plan), plan.nodes);
+      // uses full index, therefore no group sort necessary
+      assertFalse(sort_node_does_a_group_sort(plan), plan.nodes);
+      assertEqual(expected_results(query, collection.name()), execute(query, collection.name()), query);
 
-      // TODO all sorted index fields desc should work (currently does not work)
-      plan = query_plan("FOR doc IN @@collection SORT doc.a DESC, doc.x ASC RETURN doc", collection.name());
-      assertFalse(query_plan_uses_index_for_sorting(plan), plan.rules); // TODO should be true
-      // assertTrue(sort_node_does_a_group_sort(plan), plan.nodes); // TODO should work
+      query = "FOR doc IN @@collection FILTER doc.c > null AND doc.d > null SORT doc.c, doc.x RETURN doc.b";
+      plan = query_plan(query, collection.name());
+      assertTrue(query_plan_uses_index_for_sorting(plan), plan.rules);
+      assertTrue(sort_node_does_a_group_sort(plan), plan.nodes);
+      assertEqual(expected_results(query, collection.name()), execute(query, collection.name()), query);
 
-      // combined desc and asc in index fields should not work
-      plan = query_plan("FOR doc IN @@collection SORT doc.a ASC, doc.b DESC, doc.x DESC RETURN doc", collection.name());
+      query = "FOR doc IN @@collection FILTER doc.c > null SORT doc.c, doc.x RETURN doc.b";
+      plan = query_plan(query, collection.name());
+      // does not assure that doc.d != null, therefore cannot use index
       assertFalse(query_plan_uses_index_for_sorting(plan), plan.rules);
+      assertEqual(expected_results(query, collection.name()), execute(query, collection.name()), query);
+    },
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief test index usage
+    ////////////////////////////////////////////////////////////////////////////////
+
+    test_sparse_index_works_with_different_sort_directions: function () {
+      let randomNumber = randomNumberGeneratorInt(seed);
+      const collection = create_collection();
+      collection.insert(Array.from({ length: 100 }, (_, index) => index).map(i => {
+        return {
+          a: i % 9,
+          x: 100 - i - 1,
+          b: Math.floor(randomNumber() * 100),
+          c: i
+        };
+      }));
```

Comment:
Same as above. When testing something with sparse indexes, there should be some documents that are excluded.